### PR TITLE
Added visual health bars and informational damage tooltips

### PR DESCRIPTION
**Pull request by ficocelliguy**

### CODE REVIEW

File: DataService.js [hunk 1]
```diff
@@ -13,7 +13,8 @@ angular.module('PathOfDamage')
               total: 0
             },
             more: {
-              placeholder: "More Damage"
+              placeholder: "More Damage",
+              values: []
```
Comment (PsychicMuffin) [inline DataService.js:17]: why did you add the values:[] to two of the tables?

File: DataService.js [hunk 2]
```diff
@@ -85,6 +87,28 @@ angular.module('PathOfDamage')
         });
       }
       return table;
+    },
+    generateHitList: function (hits) {
+      if (!hits) {
```
Comment (PsychicMuffin) [inline DataService.js]: when is hits null?

File: app.js [hunk 3]
```diff
@@ -41,15 +48,17 @@ angular.module('PathOfDamage', [])
         }
       }
     }
+    $scope.updateHits(skipStringify);
```
Comment (PsychicMuffin) [inline app.js]: we always want to skipStringify here right, because we're going to stringify on the next line

File: style.css [hunk 4]
```diff
@@ -44,6 +44,77 @@ div.damageTable {
...
+/* Tooltip arrow */
+.damageTable .damageOutput .tooltip::after {
+    content: "";
+    position: absolute;
+    bottom: 100%;
+    left: 50%;
+    margin-left: -5px;
+    border-width: 5px;
+    border-style: solid;
+    border-color: transparent transparent #555 transparent;
```
Comment (PsychicMuffin) [inline style.css]: Can condense border properties to:
`border: 5px solid transparent;`
`border-bottom-color: #555;`

File: style.css [hunk 5]
```diff
@@ -44,6 +44,77 @@ div.damageTable {
     padding-left: 5px;
 }
 
+.damageTable .damageOutput {
+    position: relative;
+    width: 9em;
+}
+.damageTable .damageOutput .taken {
```
Comment (PsychicMuffin) [inline style.css]: blank line between blocks

File: DataService.js [hunk 6]
```diff
@@ -85,6 +87,28 @@ angular.module('PathOfDamage')
...
+        return null;
+      }
+      var list = [];
+      for (var i = 0; i < hits.length; i++) {
```
Comment (PsychicMuffin) [inline DataService.js]: we want hits.length - 1 to skip the empty hit at the end right?

File: DataService.js [hunk 7]
```diff
@@ -85,6 +87,28 @@ angular.module('PathOfDamage')
...
+        list.push(hits[i].hit);
+      }
+      return list;
+    },
+    generateHitObject: function (hitList) {
+      if (!hitList) {
```
Comment (PsychicMuffin) [inline DataService.js]: what if we restructured to if (hitList) { ...method }. Does not returning anything break anything? That makes the return undefined instead of null correct?

File: app.js [hunk 8]
```diff
@@ -80,7 +89,29 @@ angular.module('PathOfDamage', [])
...
-    return Math.round(hit || 0) + ' (' + Math.round(shifted) + ')';
+    return {
+      hit: initialHit,
+      taken: Math.round(hit || 0),
+      shifted: Math.round(shifted),
+      remaining: Math.round($scope.health - hit || 0)
+    }
+  }
+
+  $scope.updateHits = function(skipStringify) {
```
Comment (PsychicMuffin) [inline app.js:100]: I think we need to split the updateHit method up at this point. It was originally designed to be called when a hit entry was edited, which is why it checks if it needs to delete or add rows to the table. There's no real use case for looping through each hit and checking that condition for every entry. Maybe we need updateHitTable and updateHitCalc or something?

**Review by PsychicMuffin — CHANGES_REQUESTED** [review]

File: DataService.js (hunk 1 above)
Comment (ficocelliguy) [reply DataService.js:17]: you access .values in the damage calculations but you didn't initialize them, so I added it.

File: DataService.js (hunk 2 above)
Comment (ficocelliguy) [reply DataService.js]: That was from testing. Removed

File: DataService.js (hunk 6 above)
Comment (ficocelliguy) [reply DataService.js]: Yes, this was left from testing. At one point the list got shorter every refresh...

File: DataService.js (hunk 2 above)
Comment (ficocelliguy) [reply DataService.js]: This was left from testing.

File: DataService.js (hunk 7 above)
Comment (ficocelliguy) [reply DataService.js]: That will work fine, I have it use the default hits if the dataload is falsey.

File: app.js (hunk 3 above)
Comment (ficocelliguy) [reply app.js]: Right, that's cleaner.

File: app.js (hunk 8 above)
Comment (ficocelliguy) [reply app.js:100]: I originally did something like that, but then it would not properly handle the last row in the table. I think we could use a refactoring pass soon anyway, I want to for example split up the css into files and such.

File: style.css (hunk 4 above)
Comment (ficocelliguy) [reply style.css]: That looks nicer. Can you tell it wasn't my css? lol

File: DataService.js (hunk 1 above)
Comment (PsychicMuffin) [reply DataService.js:17]: None of the other tables in sections have values though, you just seemingly picked two at random to add it to? They are initialized online 163 fyi: `table.values = table.values || [];`

**Review by PsychicMuffin — APPROVED** [review]
